frontend/app/controllers/Joiner.scala

```diff
@@ -289,14 +293,20 @@ object Joiner extends Controller with ActivityTracking
       validPromotion = promotion.flatMap(_.validateFor(prpId, country).map(_ => promotion).toOption.flatten)
       destination <- request.touchpointBackend.destinationService.returnDestinationFor(request.session, request.subscriber)
       card <- paymentCard
-    } yield Ok(views.html.joiner.thankyou(
-      request.subscriber,
-      paymentSummary,
-      card,
-      destination,
-      upgrade,
-      validPromotion.filterNot(_.asTracking.isDefined)
-    )).discardingCookies(TierChangeCookies.deletionCookies: _*)
+    } yield {
+      tier match {
+        case t: Tier.Supporter if !upgrade => MembersDataAPI.Service.removeBehaviour(request)
+        case _ =>
```

Comment:
No, it's to stop the compiler emitting a warning about a non-exhaustive match if it's not there.